File: swift/Sources/TrailSamplePlugin/MetroStyle.swift
import TrailCore

/// A separate package target consuming only Trail's public interfaces.
public struct MetroStyle: TrailLineStyle {
    public let color: TrailColor
    public init(color: TrailColor = .coral) { self.color = color }
    public func draw(in context: inout TrailDrawContext) {
        context.stroke(color: .white, width: 12)
        context.stroke(color: color, width: 7)
        context.chevrons(color: .white, size: 4, spacing: 32)
    }
}
public struct QuadraticReveal: TrailAnimation {
    public init() {}
    public func sample(at time: TrailTime) -> TrailVisualState { .reveal(to: time.progress * time.progress) }
}
public extension TrailEffect {
    static func metro() -> TrailEffect { TrailEffect { Style(MetroStyle()); Animate(TrailAnimations.custom(QuadraticReveal())) } }
}

File: android/sample-plugin/src/main/kotlin/dev/trail/plugin/MetroStyle.kt
package dev.trail.plugin

import dev.trail.core.*

/** Compiled as a separate consumer module: no internal Trail APIs or registration. */
class MetroStyle(private val color: TrailColor = TrailColor.Coral) : TrailLineStyle {
    override fun draw(context: TrailDrawContext) {
        context.stroke(TrailColor.White, width = 12.0)
        context.stroke(color, width = 7.0)
        context.chevrons(TrailColor.White, size = 4.0, spacing = 32.0)
    }
}

class QuadraticReveal : TrailAnimation {
    override fun sample(time: TrailTime) = TrailVisualState.reveal(time.progress * time.progress)
}

fun TrailEffectScope.metro(color: TrailColor = TrailColor.Coral) {
    style(MetroStyle(color))
    animation(TrailAnimations.custom(QuadraticReveal()))
}
